app/scripts/modules/kubernetes/src/v2/pipelines/stages/runJob/KubernetesV2RunJobStageConfig.tsx

```diff
@@ -43,9 +61,94 @@ export class KubernetesV2RunJobStageConfig extends React.Component<IStageConfigP
     this.initRawManifest();
   }
 
+  public sourceChanged = (event: any) => {
+    this.props.updateStageField({ consumeArtifactSource: event.target.value });
+  };
+
+  public updateArtifactId(artifactId: string) {
+    this.props.updateStageField({ consumeArtifactId: artifactId });
+  }
+
+  public updateArtifactAccount(artifactAccount: string) {
+    this.props.updateStageField({ consumeArtifactAccount: artifactAccount });
+  }
+
+  public onArtifactSelected = (artifact: IExpectedArtifact) => {
+    this.props.updateStageField({ consumeArtifactId: artifact.id });
+  };
+
+  public onArtifactEdited = (artifact: IArtifact) => {
+    this.props.updateStageField({
+      consumeArtifact: artifact,
+      consumeArtifactId: artifact.id,
+      consumeArtifactAccount: artifact.artifactAccount,
+    });
+  };
+
+  public updatePropertyFile = (event: any) => {
+    this.props.updateStageField({ propertyFile: event.target.value });
+  };
+
+  private checkFeatureFlag(flag: string): boolean {
+    return !!SETTINGS.feature[flag];
+  }
+
+  public logSourceForm() {
+    const { stage } = this.props;
+    return (
+      <StageConfigField label="Container Name" helpKey="kubernetes.runJob.captureSource.containerName">
+        <input
+          className="form-control input-sm"
+          type="text"
+          value={stage.propertyFile}
+          onChange={this.updatePropertyFile}
+        />
+      </StageConfigField>
+    );
+  }
+
+  public artifactRewriteForm() {
+    const { stage, pipeline } = this.props;
+    return (
+      <StageConfigField label="Artifact">
+        <StageArtifactSelector
+          pipeline={pipeline}
+          stage={stage}
+          artifact={stage.consumeArtifact}
+          excludedArtifactTypePatterns={[]}
+          expectedArtifactId={stage.consumeArtifactId}
+          onExpectedArtifactSelected={this.onArtifactSelected}
+          onArtifactEdited={this.onArtifactEdited}
+        />
+      </StageConfigField>
+    );
+  }
+
+  public artifactForm() {
+    const { stage, pipeline } = this.props;
+    return (
+      <PreRewriteStageArtifactSelector
+        excludedArtifactTypes={[]}
+        stage={stage}
+        pipeline={pipeline}
+        selectedArtifactId={stage.consumeArtifactId}
+        selectedArtifactAccount={stage.consumeArtifactAccount}
+        setArtifactAccount={(artifactAccount: string) => this.updateArtifactAccount(artifactAccount)}
+        setArtifactId={(artifactId: string) => this.updateArtifactId(artifactId)}
+      />
+    );
+  }
+
   public render() {
     const { application, stage } = this.props;
 
+    let outputSource = <div />;
+    if (stage.consumeArtifactSource === 'propertyFile') {
+      outputSource = this.logSourceForm();
+    } else if (stage.consumeArtifactSource === 'artifact') {
+      outputSource = this.checkFeatureFlag('artifactsRewrite') ? this.artifactRewriteForm() : this.artifactForm();
```

Comment:
This is more a note for me, but wondering if you think it would be useful to create a component that abstracts away the feature flag check and conditional rendering of pre and post rewrite artifacts?

Reply:
(Not at all a note for this PR, just something I might add later)